src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ParameterTransformerConvention.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Mvc.Core.ApplicationModels
+{
+    public sealed class RouteTokenTransformerConvention : IApplicationModelConvention
+    {
+        private readonly IParameterTransformer _parameterTransformer;
+
```

Comment:
Consider implementing `IActionModelConvention`, unsealing, and adding a `protected virtual bool ShouldApply(...)` method.  Similar to https://github.com/aspnet/Mvc/pull/8390

Reply:
This lets users subclass the convention to apply it or not apply it based on some criteria they define.